pkg/ovs/deployment.go

```diff
@@ -0,0 +1,164 @@
+	}
+
+	envVars := map[string]env.Setter{}
+	envVars["KOLLA_CONFIG_FILE"] = env.SetValue(KollaConfigAPI)
+	envVars["KOLLA_CONFIG_STRATEGY"] = env.SetValue("COPY_ALWAYS")
+	envVars["CONFIG_HASH"] = env.SetValue(configHash)
+
+	//TODO(slaweq): here I need to define 3 containers
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceName,
+			Namespace: instance.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Replicas: &instance.Spec.Replicas,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: ServiceAccountName,
+					HostNetwork:        instance.Spec.HostNetwork,
+					Containers: []corev1.Container{
+						// ovsdb-server container
+						{
+							Name: ServiceName + "db-server",
+							Command: []string{
+								"/usr/bin/start-ovs",
+								"ovsdb-server",
+							},
+							Args:  args,
+							Image: instance.Spec.ContainerImage,
+							SecurityContext: &corev1.SecurityContext{
+								Capabilities: &corev1.Capabilities{
+									Add:  []corev1.Capability{"NET_ADMIN", "SYS_ADMIN", "SYS_NICE"},
+									Drop: []corev1.Capability{},
+								},
+								RunAsUser: &runAsUser,
+							},
+							Env:           env.MergeEnvs([]corev1.EnvVar{}, envVars),
+							VolumeMounts:  GetOvsDbVolumeMounts(),
+							Resources:     instance.Spec.Resources,
+							LivenessProbe: ovsDbLivenessProbe,
+						}, {
+							// ovs-vswitchd container
+							Name: ServiceName + "-vswitchd",
+							Command: []string{
+								"/usr/bin/start-ovs",
+								"ovs-vswitchd",
+							},
+							Args:  args,
+							Image: instance.Spec.ContainerImage,
+							SecurityContext: &corev1.SecurityContext{
+								Capabilities: &corev1.Capabilities{
+									Add:  []corev1.Capability{"NET_ADMIN", "SYS_ADMIN", "SYS_NICE"},
+									Drop: []corev1.Capability{},
+								},
+								RunAsUser: &runAsUser,
+							},
+							Env:           env.MergeEnvs([]corev1.EnvVar{}, envVars),
+							VolumeMounts:  GetVswitchdVolumeMounts(),
+							Resources:     instance.Spec.Resources,
+							LivenessProbe: ovsVswitchdLivenessProbe,
+						},
+						// TODO(slaweq): ovn-controller container
+					},
+				},
+			},
+		},
+	}
+	deployment.Spec.Template.Spec.Volumes = GetVolumes(instance.Name)
+	// If possible two pods of the same service should not
+	// run on the same worker node. If this is not possible
+	// the get still created on the same worker node.
+	deployment.Spec.Template.Spec.Affinity = affinity.DistributePods(
+		common.AppSelector,
+		[]string{
+			ServiceName,
+		},
+		corev1.LabelHostname,
+	)
+	if instance.Spec.NodeSelector != nil && len(instance.Spec.NodeSelector) > 0 {
+		deployment.Spec.Template.Spec.NodeSelector = instance.Spec.NodeSelector
+	}
+
+	// TODO(slaweq): Do I really need InitContainer? Maybe this can be done somehow differently?
```

Comment:
In the other operators we have used an init container to merge user-provided conf with the default conf in the service's container image.  If you don't need to do something like that, then perhaps you can exclude an init container here.  Though I did see some commands being executed in the `init.sh` you have included in this PR.

Reply:
In other operators from what I understand init container is indeed used to configure OpenStack services. In ovs-operator case this is  bit more complicated as ovn-controller is configured via external-ids stored in the ovsdb. So we need to have ovsdb-server up and running first to be able to configure ovn-controller. And because of that, init container will not work IMO. I'm removing init container from this operator, at least for now.